### PR TITLE
Feature: Added forgot password

### LINKED ISSUE #187: Forgot password not working.
Steps for reproduction

1. Try login in using confirmation email
2. Enter invalid password
3. click on forgot password

Outcome:
Blank page on `https://app.tooljet.io/forgot-password`

### PR DESCRIPTION
**Pull request by apoorv1316**
Fixes #187 

https://www.loom.com/share/c4e5b523f5de428ab8b0be92449fea07

### CODE REVIEW

**Comment by apoorv1316** [comment]
@Navaneeth-pk Please have a look. This works in development for now. I was not sure about the production config.

File: app/controllers/forgot_password_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class ForgotPasswordController < ApplicationController
+  skip_before_action :authenticate_request
+
+  def forgot
+    user = User.find_by(email: params[:_json])
+    if user.present?
+      user.send_password_reset
+      render json: {message: 'We have sent you a password reset email'}, status: :ok
```
Comment (Navaneeth-pk) [inline app/controllers/forgot_password_controller.rb:8]: Please change the text to "We've sent the confirmation code to your email address".

File: app/controllers/forgot_password_controller.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    else
+      render json: {error: 'Email address not found. Please check the email'}, status: :not_found
```
Comment (Navaneeth-pk) [inline app/controllers/forgot_password_controller.rb:10]: Please change the text to 'Email address is not associated with a ToolJet cloud account.`

File: app/mailers/user_mailer.rb [hunk 3]
```diff
@@ -11,4 +11,9 @@ def new_signup_email
     @url = "#{ENV.fetch('TOOLJET_HOST')}/invitations/#{@user.invitation_token}?signup=true"
     mail(to: @user.email, subject: 'ToolJet Invitation')
   end
+
+  def password_reset(user)
+    @user = user
+    mail to: user.email, subject: "Password reset"
```
Comment (Navaneeth-pk) [inline app/mailers/user_mailer.rb:17]: The subject should be "Password reset instructions"

File: db/migrate/20210609020556_add_forgot_password_to_user.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class AddForgotPasswordToUser < ActiveRecord::Migration[6.1]
+  def change
+    add_column :users, :forgot_password_token, :string
+    add_column :users, :forgot_password_sent_at, :datetime
```
Comment (Navaneeth-pk) [inline db/migrate/20210609020556_add_forgot_password_to_user.rb:4]: It should be `forgot_password_token_sent_at`.

**Pull request merged**